src/_langs/en/updates/2015-07-22-access-ble-devices-with-web-bluetooth.markdown

```diff
@@ -0,0 +1,279 @@
+.then(device => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+If your Bluetooth GATT Service is not the list of [the officially adopted Bluetooth
+GATT
+services](https://developer.bluetooth.org/gatt/services/Pages/ServicesHome.aspx)
+though, you may want to use the handy `window.BluetoothUUID.getService` helper
+function then. Calling it with `0x1234` will return
+`"00001234-0000-1000-8000-00805f9b34fb"`.
+
+{% highlight javascript %}
+var myService = BluetoothUUID.getService(0x1234);
+navigator.bluetooth.requestDevice({ filters: [{ services: [myService] }] })
+.then(device => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+### Connect to a Bluetooth Device
+
+So what do you do now that you have a `BluetoothDevice` resolved by the
+`navigator.bluetooth.requestDevice` Promise? Let's connect to the Bluetooth
+GATT Remote Server which holds the Attribute Protocol (ATT) lookup data and
+service and characteristic definitions.
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => {
+  // Human-readable name of the device.
+  console.log(device.name);
+  // Indicates whether or not the device is paired with the system.
+  console.log(device.paired);
+  // UUIDs of protocols, profiles and services advertised by the device.
+  console.log(device.uuids);
+
+  // Attempts to connect to GATT Remote Server.
+  return device.connectGATT();
+})
+.then(server => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+
+### Read a Bluetooth Characteristic
+
+Here we are connected to the GATT Remote Server of the Bluetooth device. Now we
+want to get a Primary Bluetooth GATT Service and read a characteristic that
+belongs to this service. Let's try for instance to read the current charge
+level of a connected device battery.
+
+In the example below, `battery_level` is the [adopted Bluetooth GATT
+Characteristic Battery
+Level](https://developer.bluetooth.org/gatt/characteristics/Pages/CharacteristicViewer.aspx?u=org.bluetooth.characteristic.battery_level.xml).
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => device.connectGATT())
+.then(server => {
+  // Getting Battery Service...
+  return server.getPrimaryService('battery_service');
+})
+.then(service => {
+  if (!service) throw 'Battery Service not found';
+  // Getting Battery Level Characteristic...
+  return service.getCharacteristic('battery_level');
+})
+.then(characteristic => {
+  // Reading Battery Level...
+  return characteristic.readValue();
+})
+.then(buffer => {
+  var data = new DataView(buffer);
+  console.log('Battery percentage is ' + data.getUint8(0));
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+If you use a custom Bluetooth GATT characteristic,
+`window.BluetoothUUID.getCharacteristic` definitely comes in handy. Calling it
```

Comment:
In this case getCharacteristic is just calling BluetoothUUID.canonicalUUID. Also this is implicit in service.getCharacteristic, so we can probably simplify or cut this section.